### PR TITLE
fix(dragdrop): Reordering item out of LV items bounds

### PR DESCRIPTION
**Pull request by dr1rrb**
closes https://github.com/unoplatform/nventive-private/issues/329

## Bugfix
Reordering items before or after the bounds of items on the LV has no effect.

## What is the current behavior?
* When we leave the bounds of a drop target, we don't reset the `dragUIOverride` object, so the accepted operation (i.e. `None`) is not updated in the `DragView`.
* In a `ListView` when reordering items after the bounds of the last item, the accepted operation remains to `Move` (as expected)  but drop has no effect.

## What is the new behavior?
* On pointer move, if we are over a new drop target, or if we are not able to find a drop target, we clear the `dragUIOvwerride`
* If we are not able to find an item under the pointer when reordering items of a `ListView`, we are selecting the last one.

## PR Checklist
- [ ] ~~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs [hunk 1]
```diff
@@ -213,17 +210,25 @@ private static void OnReorderUpdated(object sender, _DragEventArgs dragEventArgs
...
+		}
+
+		private static void OnReorderDragLeave(object sender, _DragEventArgs dragEventArgs)
+		{
+			var that = sender as ListView;
+			var src = dragEventArgs.DataView.FindRawData(ReorderOwnerFormatId) as ListView;
+			if (that is null || src != that)
 			{
-				that.SetPendingAutoPanVelocity(PanVelocity.Stationary);
+				dragEventArgs.Log().Warn("Invalid reorder event.");
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs:225]: use if(this.Log().IsEnabled...)

**Review by carldebilly — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs [hunk 2]
```diff
@@ -1506,13 +1511,25 @@ private void UpdateLayoutForReordering()
 			if (_reorderingState is { } reorderingState && reorderingState.LayoutAttributes is { } draggedAttributes)
 			{
 				var dropTargetAttributes = GetLayoutAttributesUnderPoint(reorderingState.Location);
+				if (dropTargetAttributes is null)
+				{
+					// If we have a reorderingState, it means that the pointer is still over the LV.
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs:1516]: Could it also be before the first item?

**Review by davidjohnoliver — COMMENTED** [review]
This is already handled properly on Android and WASM?

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs:1516]: Do we support to vertically center items in a ListView?

**Comment by dr1rrb** [comment]
> This is already handled properly on Android and WASM?

From my tests, yes. It was only on iOS

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs (hunk 2 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs:1516]: not to my knowledge, but what if, eg, there's a Padding set on the ListView or its ItemsPresenter?

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs]: Good point ... fixed!

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs [hunk 3]
```diff
@@ -1578,20 +1583,33 @@ private void ResetReorderedLayoutAttributes()
...
+
 			foreach (var dict in _itemLayoutInfos.Values)
 			{
 				foreach (var layoutAttributes in dict.Values)
 				{
-					if (DoesLayoutAttributesContainDraggedPoint(point, layoutAttributes))
+					var distance = ((Rect)layoutAttributes.Frame).GetDistance(adjustedPoint);
+					if (distance == 0)
 					{
+						// Fats path: we found the element that is under the element
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.iOS.cs:1600]:
```suggestion
						// Fast path: we found the element that is under the element
```

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**